exchange/exchange.go

```diff
@@ -136,6 +135,10 @@ func (e *exchange) HoldAuction(ctx context.Context, r AuctionRequest, debugLog *
 	}
 
 	debugInfo := getDebugInfo(r.BidRequest, requestExt)
+
+	debugInfo = debugInfo && r.Account.DebugAllowed
+	debugLog.Enabled = debugLog.Enabled && r.Account.DebugAllowed
```

Comment:
`debugLog` can be nil. This needs a nil check, or perhaps incorporate the `r.Account.DebugAllowed` condition in line 203 and 228. Could you please add a unit test to cover when `debugLog` is nil, because that doesn't seem to have coverage.

Reply:
Good catch! Also modified existing test to make sure `debugLog` is properly handled in Auction function.